php/oop/assignments/zoo.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+abstract class Animal
+{
+    public function __construct(private string $name, private bool $is_parturition)
+    {
+    }
+
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    
+    public function sound()
+    {
+        return 'my sound is like ' . $this->name;
+    }
+
+    public function move()
+    {
+        return 'I can move like ' . $this->name;
+    }
+}
+
+class WildAnimal extends Animal
+{
+    public function hunt(Animal $animal)
+    {
+        if (is_subclass_of($animal, 'WildAnimal')) {
```

Comment:
I suggest `WildAnimal::class` instead of `WildAnimal`

Reply:
Hi 
I change the requested part and also add parturition method by  using trait but i don't know this method is correct or not.